commit 6c8069b7d18390ff854f480180394a56769df461 (refs/changes/49/4849/1)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-09-18T20:57:20+00:00 (1 year, 1 month ago)
    
    MANTA-3950 muskie_deleted_bytes metric should look at the creator of an object instead of the caller

diff --git a/lib/audit.js b/lib/audit.js
index 59a5d27..3fdc2b7 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -300,26 +300,23 @@ function auditLogger(options) {
         }
 
         if (op === 'DELETE') {
+            var md = req.metadata;
+            var owner = md.creator || md.owner;
+
             /*
-             * Delete metrics for snaplink-disabled accounts provide a
-             * meaningful measure of short-term storage reclamation in
-             * deployments with garbage-collectors.
-             *
-             * We avoid tracking these metrics for accounts that are not
-             * snaplink-disabled because the storage for objects owned by such
-             * accounts may still be occupied for long periods of time after
-             * the corresponding delete API calls complete.
+             * Count bytes for which DELETE API requests have been completed. If
+             * the owner of the underlying data has snaplinks disabled, then
+             * the deleted object was processed by the accelerated deletion
+             * pipeline.
              */
-            common.checkAccountSnaplinksEnabled(req, req.caller.account.uuid,
+            common.checkAccountSnaplinksEnabled(req, owner,
                 function (enabled) {
-                if (enabled) {
-                    return;
-                }
-                var md = req.metadata;
                 if (md.type === 'object' && md.contentLength > 0) {
                     var storage = md.contentLength * md.sharks.length;
-                    labels.login = req.caller.account.login ||
-                        req.caller.account.uuid;
+                    labels = {
+                        accelerated_gc: enabled,
+                        owner: owner
+                    };
                     deleted_data_counter.add(storage, labels);
                 }
             });
